pkg/operator/controller/dnsnameresolver/controller.go

```diff
@@ -0,0 +1,217 @@
+package dnsnameresolver
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	networkv1alpha1 "github.com/openshift/api/network/v1alpha1"
+
+	"github.com/sirupsen/logrus"
+
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ipRemovalGracePeriod = time.Second * 5
+
+	controllerName = "dnsnameresolver_controller"
+)
+
+// NewUnmanaged creates and returns a controller that watches DNSNameResolver
+// objects. The controller re-resolves the DNS names which get added to the
+// status of the DNSNameResolver objects. It also removes IP addresses of DNS
+// names, whose TTLs have expired, from the staus. This is an unmanaged
```

Comment:
```suggestion
// names, whose TTLs have expired, from the status. This is an unmanaged
```

Reply:
Moved to openshift/coredns-ocp-dnsnameresolver#2